library/modules/bigip_gtm_dns_listener.py

```diff
@@ -0,0 +1,943 @@
+
+    @property
+    def disabled_vlans(self):
+        return self.vlan_status
+
+    @property
+    def vlan_status(self):
+        result = dict()
+        vlans = self.vlans
+        if vlans is not None:
+            result['vlans'] = vlans
+        self._update_vlan_status(result)
+        return result
+
+    @property
+    def irules(self):
+        if self.want.irules is None:
+            return None
+        if self.want.irules == '' and len(self.have.irules) > 0:
+            return []
+        if self.want.irules in [[], ''] and len(self.have.irules) == 0:
+            return None
+        if self.want.irules != self.have.irules:
+            return self.want.irules
+
+
+class ModuleManager(object):
+    def __init__(self, *args, **kwargs):
+        self.module = kwargs.get('module', None)
+        self.client = F5RestClient(**self.module.params)
+        self.want = ModuleParameters(params=self.module.params)
+        self.have = ApiParameters()
+        self.changes = UsableChanges()
+
+    def _set_changed_options(self):
+        changed = {}
+        for key in Parameters.returnables:
+            if getattr(self.want, key) is not None:
+                changed[key] = getattr(self.want, key)
+        if changed:
+            self.changes = UsableChanges(params=changed)
+
+    def _update_changed_options(self):
+        diff = Difference(self.want, self.have)
+        updatables = Parameters.updatables
+        changed = dict()
+        for k in updatables:
+            change = diff.compare(k)
+            if change is None:
+                continue
+            else:
+                if isinstance(change, dict):
+                    changed.update(change)
+                else:
+                    changed[k] = change
+        if changed:
+            self.changes = UsableChanges(params=changed)
+            return True
+        return False
+
+    def _announce_deprecations(self, result):
+        warnings = result.pop('__warnings', [])
+        for warning in warnings:
+            self.client.module.deprecate(
+                msg=warning['msg'],
+                version=warning['version']
+            )
+
+    def _override_vlans_if_all_specified(self):
+        '''Overrides any specified VLANs if 'all' VLANs are specified
+
+        The special setting 'all VLANs' in a BIG-IP requires that no other VLANs
+        be specified. If you specify any number of VLANs, AND include the 'all'
+        VLAN, this method will erase all of the other VLANs and only return the
+        'all' VLAN.
+        '''
+        all_vlans = ['/common/all', 'all']
+        if self.want.enabled_vlans is not None:
+            if any(x for x in self.want.enabled_vlans if x.lower() in all_vlans):
+                self.want.update(
+                    dict(
+                        enabled_vlans=[],
+                        vlans_disabled=True,
+                        vlans_enabled=False
+                    )
+                )
```

Comment:
this would be better suited to be done in ModuleParameters rather then here?

Reply:
I agree - this is better suited to be done in ModuleParameters. I have made the change and this comment is addressed now